src/anyp/TrafficMode.h

```diff
@@ -58,27 +67,99 @@ class TrafficMode
      *  - URL translation from relative to absolute form
      *  - Same-Origin verification is mandatory
      *  - destination pinning is recommended
-     *  - authentication prohibited
+     *  - Squid authentication prohibited
      */
     bool tproxyIntercept = false;
+};
+
+/**
+ * Set of 'mode' flags defining types of traffic which can be received.
+ *
+ * Use to determine the processing steps which need to be applied
+ * to this traffic under any special circumstances which may apply.
+ */
+class TrafficMode
+{
+public:
+    /// This port handles traffic that has been intercepted prior to being delivered
+    /// to the TCP client of the accepted connection and/or to us. This port mode
+    /// alone does not imply that the client of the accepted TCP connection was not
+    /// connecting directly to this port (since commit 151ba0d).
+    bool interceptedSomewhere() const { return flags_.natIntercept || flags_.tproxyIntercept || flags_.proxySurrogateHttps; }
+
+    /// The client of the accepted TCP connection was connecting to this port.
+    /// The accepted traffic may have been intercepted earlier!
+    bool tcpToUs() const { return proxySurrogate() || !interceptedSomewhere(); }
```

Comment:
Part of my issue with this member has been resolved now with changes to interceptedSomewhere().

However, I still dislike its name in regard to how proxySurrogate() can be true. The current implementation of PROXY protocol completely replaces the tcpClient IP:port details for the connection. Whereas the "tcpToUs" implies that we can use the xaction::tcpClient object as the TCP details for frontend<->squid connection.

Fixing that situation is outside scope, but I would like to see the member name used here be clear about the truth rather than hide it behind abstraction.

Reply:
> I still dislike its name

The current name is the best name suggested so far. It is difficult for me to suggest alternatives because I very much disagree with your assessment of the current naming implications/problems (but I would rather not spend time arguing about those if a different name would make you happy). Would you prefer transportToUs() instead of tcpToUs()? Any better naming ideas?

Reply:
The RFC7230 term for this type of proxy service of a client is "explicit proxy". It covers only the immediate client->Squid knowledge, so proxySurrogate is true here while technically being intercepted (somewhere else).

IMO, if we have to have it at all the name should be explicitProxy(). That said, other members here should not be defined in terms of this one - precisely because of how it is limited to ignoring proxySurrogate() details about remote clients. Thus my change requests on how interceptedLocally() etc are defined.

Reply:
> The RFC7230 term for this type of proxy service of a client is "explicit proxy".

RFC 7230 does not use the term "explicit proxy". It calls explicit proxies "proxy" (see Section [2.3](https://datatracker.ietf.org/doc/html/rfc7230#section-2.3)). More importantly, the term "explicit proxy" excludes "reverse proxy" (a.k.a. "gateway"). The tcpToUs() method we are discussing returns true for reverse proxies, so we cannot call it an explicitProxy().

> other members here should not be defined in terms of this one - precisely because of how it is limited to ignoring proxySurrogate() details about remote clients

The method in question -- tcpToUs() -- is a test of a specific (documented) relationship between the client of the accepted connection and the Squid port. I do not understand why anything else should matter to this method. If a method returns correct result while ignoring x, the method should continue to ignore x. Why should not it?!

PR methods were very logical:

```C++
bool interceptedRemotely() const { return interceptedSomewhere() && tcpToUs(); }
bool interceptedLocally() const { return interceptedSomewhere() && !tcpToUs(); }
```

Is there a different way to express the same truth tables? Sure! There are many. I have given up my attempts to explain why these methods are better than the alternatives and inlined them in commits 297c72d and 1342ef4 (Eduard has beaten me by a few minutes, but his changes did not go far enough to completely remove these methods, and I did not want to waste more time on these debates).

Please note that while we could remove internally-used (and unused) methods, we cannot remove interceptedSomewhere() because it is used externally, in many places.